### PR TITLE
[Feature] GCKey Sign Up Page

### LINKED ISSUE #7690: ✨ Sign up using GCkey
## ✨ Sign up using GCKey

This page will replace the [current GCKey sign up page](https://talent.canada.ca/en/register-info). 

The idea is that this is give the users more context and aids before we send them off to GCKey.

## 🎨 Figma file
[Figma link](https://www.figma.com/file/GGyd3OuIpYxHBydvx65Fk2/Candidates---GCKey-prep-pages?type=design&node-id=109-7373&mode=design&t=5WFdEmxhChielXb2-4)

### 🧑‍🎨 Designer

@BrookeCameron 

## 📸 Screenshot
Screenshot 2023-08-25 at 9 13 19 AM

### 🧩 Component 
Regular page header with graphic 

### ✍🏼 Copy 
H1: Sign up using GCKey 

### ✍🏼 Copy 
H2: What to expect when creating an account 
Paragraph: 
- To sign up (or authenticate) with GC Digital Talent, we require that you use GCKey.
- GCKey is a Government of Canada credential that allows you to securely conduct online business with various governmental programs and services.    
- In addition to GCKey, you will need to set up a two-factor authenticator app for an added layer of security. (See below for details)
- Expect to spend approximately 20 minutes to set up GCKey and the authenticator app.
- We have prepared additional guidance In the next section to help.

### 🎯 Button 
Button copy: Continue to GCKey and sign up 
Button link: https://clegc-gckey.gc.ca/j/eng/l?ReqID=_1a539bc49ac8d7c8b2d4 

### ✍🏼 Copy 
H3: Creating a GCKey account
H4: Steps to create your GCKey account 

#### Step 1
Image: 
Gckey_illustration_blocks_V1_Sign up
Image alt: mark as decorative 
Paragraph copy:  1. Select **sign up** on the welcome page. This is **located after the sign in** section. 

#### Step 2
Image: 
Gckey_illustration_blocks_V1_username and pass
Image alt: mark as decorative 
Paragraph copy: 2. Create a username and password. Don’t forget to **save your username** separately from your **email address**.

#### Step 3
Image: 
Gckey_illustration_blocks_V1_Recovery Qs
Image alt: mark as decorative 
Paragraph copy: 3. Complete your **recovery questions**. You are required to use a memorable **person** and **date**.

#### Step 4
Image: 
Gckey_illustration_blocks_V1_Email code
Image alt: mark as decorative 
Paragraph copy:  4. Complete the email verification process. **A unique one time code** will be emailed to you to enter into GCKey. 

### ✍🏼 Copy 
H3: Part 2: Setting up two-factor authentication

H4:  Why set up a two-factor authentication? 
Paragraph: Think of the sign in as a security door to your account that can only be opened with a key (your password) and a secret code (from your two-factor authentication app). Even if someone finds or copies your key, they still won’t be able to get in because they don’t have your secret code. 

H4: You will need an authenticator app. 
Paragraph: If you don’t already have an authenticator app you will need to download one. Digital vendors, like Google Authenticator and Microsoft Authenticator, provide authenticator apps. Whichever app you choose, ensure that it comes from a reputable vendor. 

H4: Setting up your two-factor authentication app

#### Step 1
Image: 
Gckey_illustration_blocks_V1_Download app
Image alt: mark as decorative 
Paragraph copy: 1. **Prepare your device** by downloading your preferred two-factor authentication app

#### Step 2
Image: 
Gckey_illustration_blocks_V1_Scan QR code
Image alt: mark as decorative 
Paragraph copy:  2. **Scan the QR code** or **enter your secret code** using your two-factor authenticator app.

#### Step 3
Image: 
Gckey_illustration_blocks_V1_6 digit code
Image alt: mark as decorative 
Paragraph copy: 3. **Enter** the code generated **from your authenticator app** into the verification bar.

#### Step 4
Image: 
Gckey_illustration_blocks_V1_Hooray
Image alt: mark as decorative 
Paragraph copy: 4. Hooray! You’ve completed your GCKey account and will be **returned to the GC Digital Talent platform.**

### 🧩 Component 
Expansion block simple

### ✍🏼 Copy 

**What is a GCKey?**
A GCKey is a central credential not managed by the GC Digital Talent team. The Government of Canada offers it as a way for you to communicate securely with many online-enabled Government programs and services.

**Who do I contact if I have questions about GCKey?**
If you have questions about GCKey, please contact the GCKey team at:
Canada and the United States
1-855-438-1102
Text Telephone (TTY/TDD)
1-855-438-1103
Outside Canada and the United States
1-800-2318-6290
Customer Service Representatives are available to assist you by phone,
year round, 24 hours a day, 7 days a week.

**Which authenticator app should I install?**
As the Government of Canada we cannot recommend any specific third-party vendors or apps. Well known digital vendors, like Google Authenticator and Microsoft Authenticator, provide authenticator apps. Whichever app you choose, ensure that it comes from a reputable vendor. 

**What if I deleted the app or changed phone and I don’t have the recovery codes?**
Although your login cannot be recovered, you can contact our [Help Desk](https://talent.canada.ca/en/support), and they can help you recover your account.

**Can I use SMS or email authentication instead of an app?**
Currently, our site only supports authentication through an authenticator app.

**Can you remove the two-factor authentication from my account so I can reset it?**
We cannot remove the two factor authentication from your account, but you can contact our [Help Desk](https://talent.canada.ca/en/support) and they can assist you with account recovery.

**What can I do if my authenticator codes are no longer being accepted?**
Please contact our [Help Desk](https://talent.canada.ca/en/support), and they can help you recover your account.

**Already have a GCKey account?**
If you already have a GCKey account you can sign in to your GC Digital Talent profile using your existing GCKey, even if you've never used this platform before.
If you're unsure whether you have an existing GCKey account, continue to the website and try signing in. If you can't remember your password, you can also reset it there.

Read all the FAQ’s and still stuck?  [Contact our team for help](https://talent.canada.ca/en/support)

## bottom of page buttons 

[Sign in instead](https://talent.canada.ca/en/login-info) 

[Continue to GCKey and sign up](https://clegc-gckey.gc.ca/j/eng/l?ReqID=_7d9e75dbf7bf2ee0af67)

## ✅ Acceptance Criteria

- [ ] Page matches Figma design
- [ ] Page exists an expected URL
- [ ] Storybook story
- [ ] Review from key designer

### PR DESCRIPTION
**Pull request by petertgiles**
🤖 Resolves #7690 

## 👋 Introduction

This branch includes a new look for the Sign Up page.

## 🕵️ Details

- The special IAP view still exists and has not been modified.  You may want to review the diff with whitespace changes hidden.
image

## 🧪 Testing

1. Rebuild the site
2. Visit `http://localhost:8000/en/register-info`

## 📸 Screenshot

image

### CODE REVIEW

**Comment by BrookeCameron** [comment]
Same comments about font weights and overall spacing as #7835

File: packages/ui/src/components/Heading/styles.ts
Comment (vd1992) [inline packages/ui/src/components/Heading/styles.ts:1]: Changes here seems to be kicking up a bunch of Chromatic chatter

image

Some of it nigh-unnoticeable

Some of it pretty noticeable

image

image

Comment (petertgiles) [reply packages/ui/src/components/Heading/styles.ts:1]: Yep, this was from the designer review.
image

@BrookeCameron You're sure you want this change?

Comment (JamesHuf) [reply packages/ui/src/components/Heading/styles.ts:1]: This has been reduced in #7835 now.

Comment (petertgiles) [reply packages/ui/src/components/Heading/styles.ts:1]: Thanks James!  I've grabbed it.
[thinner icons](https://github.com/GCTC-NTGC/gc-digital-talent/pull/7834/commits/0814635466af547f6488ba94dad5ec1892938f63)

File: apps/web/src/pages/Auth/SignUpPage/SignUpPage.tsx [hunk 1]
```diff
@@ -54,147 +120,680 @@ const SignUpPage = () => {
...
+          </>
+        )}
+
+        <hr data-h2-margin="base(x2, 0)" />
+        <div
+          data-h2-display="base(flex)"
+          data-h2-flex-direction="base(column) l-tablet(row)"
+          data-h2-align-items="base(center)"
+          data-h2-justify-content="base(space-between)"
+        >
```
Comment (esizer) [inline apps/web/src/pages/Auth/SignUpPage/SignUpPage.tsx:778]: We should add either a gap value here or margin to the paragraph tags. Looks a little tight on mobile.

Screenshot 2023-09-08 150909

File: apps/web/src/pages/Auth/SignUpPage/SignUpPage.tsx [hunk 2]
```diff
@@ -54,147 +120,680 @@ const SignUpPage = () => {
...
-              <p>
+      <div
+        data-h2-container="base(center, large, x1) p-tablet(center, large, x2)"
+        data-h2-margin-bottom="base(x1)"
+      >
+        {!iapMode ? (
+          <>
+            {/* Standard copy */}
+            <Heading
+              Icon={MapIcon}
```
Comment (esizer) [inline apps/web/src/pages/Auth/SignUpPage/SignUpPage.tsx:131]: I feel like this was fixed at some point. Not sure if it makes sense here but the icon in our `Heading` component should have `flex-shrink: 0'` or else it looks odd on mobile.

Screenshot 2023-09-08 151156

**Review by vd1992 — APPROVED** [review]
Works for me locally and the Chromatic Diff is much more muted now, nice!
If it is also good from the design review step then good to go 💯 

Will spin off an issue for a translation thing...

**Review by esizer — CHANGES_REQUESTED** [review]
Some minor mobile stuff to take into consideration.

Also, those steps really should be ordered lists for a11y. I guess we couldn't figure that out with the images? 😢

File: apps/web/src/pages/Auth/SignUpPage/SignUpPage.tsx (hunk 1 above)
Comment (petertgiles) [reply apps/web/src/pages/Auth/SignUpPage/SignUpPage.tsx:778]: Good catch!  Contributed to the other branch.
[styling tweaks from dev review](https://github.com/GCTC-NTGC/gc-digital-talent/pull/7835/commits/8fabb396dca6bcbf0dc5e32332e2e73c3034efda)

File: apps/web/src/pages/Auth/SignUpPage/SignUpPage.tsx (hunk 2 above)
Comment (petertgiles) [reply apps/web/src/pages/Auth/SignUpPage/SignUpPage.tsx:131]: Good catch!  Contributed to the other branch.
[styling tweaks from dev review](https://github.com/GCTC-NTGC/gc-digital-talent/pull/7835/commits/8fabb396dca6bcbf0dc5e32332e2e73c3034efda)

**Pull request merged**